### PR TITLE
[tcling] Refactor code in separate routines and reduce include for modules.

### PR DESCRIPTION
**Pull request by vgvassilev**
This patch moves code which does similar things into standalone routines. This
way we make more clear the intent of use.

### CODE REVIEW

**Comment by pcanal** [comment]
minor details, in TCling.cxx the indentation is 3 characters

File: core/metacling/src/TCling.cxx [hunk 1]
```diff
@@ -1112,6 +1112,95 @@ static std::string GetModuleNameAsString(clang::Module *M, const clang::Preproce
...
+  std::string PreIncludes;
+  bool hasCxxModules = clingInterp.getCI()->getLangOpts().Modules;
+
+   // For the list to also include string, we have to include it now.
+   // rootcling does parts already if needed, e.g. genreflex does not want using
+   // namespace std.
+   if (IsFromRootCling()) {
+     PreIncludes += "#include \"RtypesCore.h\"\n";
+   } else {
+     if (!hasCxxModules)
```
Comment (pcanal) [inline core/metacling/src/TCling.cxx]: This part of the code seems to be 'new' (i.e. add support for hasCxxModules).    If this the case, it ought to be mentioned in the commit log (and the NFC should be removed).  Thanks.

Comment (vgvassilev) [reply core/metacling/src/TCling.cxx]: Yeah, ran clang-format.

Comment (pcanal) [reply core/metacling/src/TCling.cxx]: Thanks.  Can you also update the commit to mention the new functionality?

Comment (vgvassilev) [reply core/metacling/src/TCling.cxx]: Done.

**Comment by vgvassilev** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Pull request merged**